driver-core/src/main/java/com/datastax/driver/core/PagingState.java

```diff
@@ -0,0 +1,158 @@
+/*
+ *      Copyright (C) 2012-2015 DataStax Inc.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package com.datastax.driver.core;
+
+import java.io.Serializable;
+import java.nio.ByteBuffer;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Arrays;
+
+import com.datastax.driver.core.exceptions.InvalidQueryException;
+import com.datastax.driver.core.utils.Bytes;
+
+/**
+ * A Paging State object for handling pager management with a Cassandra server.
+ *
+ * Using a PagingState is useful when you want to fetch a certain number of
+ * pages of result from a Query returning a big amount of data. By saving the PagingState
+ * and reusing it further in your application, you can fetch the next page of results starting
+ * from where you had saved the PagingState (and not from the beginning).
+ *
+ * The PagingState can be serialised and deserialised. It can also be converted into
+ * a String (in Hexadecimal representation) and re-built from a previously converted String.
+ *
+ * PagingState comes with a certified hash verification which allows to certify the validity
+ * of the PagingState buffer. Hence, using a saved PagingState with another Statement than the
+ * one it was made with, will fail the verification and the saved PagingState won't be
+ * usable. Also, if the PagingState content buffer (or the String the PagingState was converted to)
+ * is modified , the verification will fail, too. The content of the PagingState should not be modified
+ * at any time.
+ *
+ */
+
+public class PagingState implements Serializable {
```

Comment:
I overlooked something: implementing `Serializable` won't work out of the box, because `ByteBuffer` is not serializable.  Let's drop this for now, exposing the raw ByteBuffer should be sufficient for most cases, and we can still come up with something later.